src/sentry/tasks/derive_code_mappings.py

```diff
@@ -69,62 +53,30 @@ def derive_code_mappings(organization_id: int, dry_run=False) -> None:
 
     trees: JSONData = installation.get_trees_for_org()
     trees_helper = CodeMappingTreesHelper(trees)
-    for project, stacktrace_paths in project_stacktrace_paths.items():
-        code_mappings = trees_helper.generate_code_mappings(stacktrace_paths)
-        set_project_codemappings(code_mappings, organization, organization_integration, project)
-
-
-def identify_stacktrace_paths(organization: Organization) -> Mapping[Project, List[str]]:
-    """
-    Generate a map of projects to stacktrace paths for an organization.
-
-    This filters out non-python projects, or projects without an event
-    in the last 7 days.
-    """
-    projects = Project.objects.filter(
-        organization=organization,
-        first_event__isnull=False,
-    )
-    projects = [
-        project
-        for project in projects
-        if Group.objects.filter(
-            project=project,
-            last_seen__gte=timezone.now() - ACTIVE_PROJECT_THRESHOLD,
-        ).exists()
-    ]
-
-    return {project: get_all_stacktrace_paths(project) for project in projects}
-
-
-def get_all_stacktrace_paths(project: Project) -> List[str]:
-    groups = Group.objects.filter(
-        project=project,
-        last_seen__gte=timezone.now() - GROUP_ANALYSIS_RANGE,
-        platform="python",
+    code_mappings = trees_helper.generate_code_mappings(stacktrace_paths)
+    set_project_codemappings(
+        code_mappings,
+        organization_integration,
+        organization_id,
+        project_id,
     )
 
-    all_stacktrace_paths = set()
-    for group in groups:
-        event = group.get_latest_event()
-        all_stacktrace_paths.update(get_stacktrace_paths(event.data))
-
-    return list(all_stacktrace_paths)
-
 
-def get_stacktrace_paths(data: NodeData) -> Set[str]:
+def identify_stacktrace_paths(data: NodeData) -> List[str]:
     """
-    Get the stacktrace_paths from the stacktrace for the latest event for an issue.
+    Get the stacktrace_paths from the event data.
     """
+    if data["platform"] != "python":
+        return []
     stacktraces = get_stacktrace(data)
     stacktrace_paths = set()
     for stacktrace in stacktraces:
         try:
-            paths = {frame["filename"] for frame in stacktrace["frames"]}
+            paths = {frame["filename"] for frame in stacktrace["frames"] if frame["in_app"]}
```

Comment:
We can discard frames that are not "in_app" for code mappings

Reply:
Can you add an extra except block for when "in_app" is not defined? I don't know if it's a mandatory field.